app/models/form_submission.rb

```diff
@@ -10,7 +10,47 @@ class FormSubmission < ApplicationRecord
 
   validates :form_type, presence: true
 
-  def latest_pending_attempt
-    form_submission_attempts.where(aasm_state: 'pending').order(created_at: :asc).last
+  class << self
+    # rubocop:disable Metrics/MethodLength
+    def with_latest_intake(user_account)
+      # This query retrieves distinct form submissions for a given user account,
+      # along with the latest benefits intake UUID for each form. If a form has any
+      # submission attempts, it uses the latest attempt's UUID; otherwise, it falls
+      # back to the form submission's original UUID.
+      FormSubmission
+        .select(
+          'DISTINCT form_submissions.id,
+          form_submissions.form_type,
+          COALESCE(latest_attempts.benefits_intake_uuid, form_submissions.benefits_intake_uuid) AS benefits_intake_uuid,
+          form_submissions.created_at'
+        )
+        .left_joins(:form_submission_attempts)
+        .joins(
+          'LEFT JOIN (
+            SELECT fsa.form_submission_id, fsa.benefits_intake_uuid
+            FROM form_submission_attempts fsa
+            INNER JOIN (
+              SELECT form_submission_id, MAX(created_at) AS latest_attempt
+              FROM form_submission_attempts
+              GROUP BY form_submission_id
+            ) latest_fsa ON fsa.form_submission_id = latest_fsa.form_submission_id
+                          AND fsa.created_at = latest_fsa.latest_attempt
+          ) latest_attempts ON form_submissions.id = latest_attempts.form_submission_id'
+        )
+        .where(user_account:)
+    end
+    # rubocop:enable Metrics/MethodLength
```

Comment:
I believe you could circumvent the `Metrics/MethodLength` complaint by pulling out the SQL, like so. If you were so inclined:

```suggestion
class << self
  def with_latest_intake(user_account)
    FormSubmission
      .select(
        'DISTINCT form_submissions.id,
        form_submissions.form_type,
        COALESCE(latest_attempts.benefits_intake_uuid, form_submissions.benefits_intake_uuid) AS benefits_intake_uuid,
        form_submissions.created_at'
      )
      .left_joins(:form_submission_attempts)
      .joins(latest_attempts_join)
      .where(user_account: user_account)
  end

  private

  def latest_attempts_join
    <<-SQL.squish
      LEFT JOIN (
        SELECT fsa.form_submission_id, fsa.benefits_intake_uuid
        FROM form_submission_attempts fsa
        INNER JOIN (
          SELECT form_submission_id, MAX(created_at) AS latest_attempt
          FROM form_submission_attempts
          GROUP BY form_submission_id
        ) latest_fsa ON fsa.form_submission_id = latest_fsa.form_submission_id
                      AND fsa.created_at = latest_fsa.latest_attempt
      ) latest_attempts ON form_submissions.id = latest_attempts.form_submission_id
    SQL
  end
end

```

Reply:
Yes.  I was pretty sure there was a better way, so thanks for the feedback on this.  I'll make the changes.